src/control/cmd/dmg/pretty/pool_test.go

```diff
@@ -209,6 +209,310 @@ Rebuild failed, rc=0, status=2
+								Total: 100000000000,
+								Free:  90000000000,
+							},
+						},
+					},
+				},
+			},
+			expPrintStr: `
+Target: type unknown, state new
+- Storage tier 0 (SCM):
+  Total size: 6.0 GB
+  Free: 5.0 GB
+- Storage tier 1 (NVMe):
+  Total size: 100 GB
+  Free: 90 GB
+`,
+		},
+		"valid: single target (unknown, drain)": {
+			pqtr: &control.PoolQueryTargetResp{
+				Status: 0,
+				Infos: []*control.PoolQueryTargetInfo{
+					{
+						Type:  0,
+						State: control.PoolTargetStateDrain,
+						Space: []*control.StorageTargetUsage{
+							{
+								Total: 6000000000,
+								Free:  5000000000,
+							},
+							{
+								Total: 100000000000,
+								Free:  90000000000,
+							},
+						},
+					},
+				},
+			},
+			expPrintStr: `
+Target: type unknown, state drain
+- Storage tier 0 (SCM):
+  Total size: 6.0 GB
+  Free: 5.0 GB
+- Storage tier 1 (NVMe):
+  Total size: 100 GB
+  Free: 90 GB
+`,
+		},
+		"valid: multiple target (mixed statuses exclude 2 targets in progress)": {
+			pqtr: &control.PoolQueryTargetResp{
+				Status: 0,
+				Infos: []*control.PoolQueryTargetInfo{
+					{
+						Type:  0,
+						State: control.PoolTargetStateDown,
+						Space: []*control.StorageTargetUsage{
+							{
+								Total: 6000000000,
+								Free:  5000000000,
+							},
+							{
+								Total: 100000000000,
+								Free:  90000000000,
+							},
+						},
+					},
+					{
+						Type:  0, // FIXME: control.PoolTargetTypeUnknown
+						State: control.PoolTargetStateUpIn,
+						Space: []*control.StorageTargetUsage{
+							{
+								Total: 6000000000,
+								Free:  5000000000,
+							},
+							{
+								Total: 100000000000,
+								Free:  90000000000,
+							},
+						},
+					},
+					{
+						Type:  0,
+						State: control.PoolTargetStateDownOut,
+						Space: []*control.StorageTargetUsage{
+							{
+								Total: 6000000000,
+								Free:  5000000000,
+							},
+							{
+								Total: 100000000000,
+								Free:  90000000000,
+							},
+						},
+					},
+					{
+						Type:  0,
+						State: control.PoolTargetStateUpIn,
+						Space: []*control.StorageTargetUsage{
+							{
+								Total: 6000000000,
+								Free:  5000000000,
+							},
+							{
+								Total: 100000000000,
+								Free:  90000000000,
+							},
+						},
+					},
+				},
+			},
+			expPrintStr: `
+Target: type unknown, state down
+- Storage tier 0 (SCM):
+  Total size: 6.0 GB
+  Free: 5.0 GB
+- Storage tier 1 (NVMe):
+  Total size: 100 GB
+  Free: 90 GB
+Target: type unknown, state up_in
+- Storage tier 0 (SCM):
+  Total size: 6.0 GB
+  Free: 5.0 GB
+- Storage tier 1 (NVMe):
+  Total size: 100 GB
+  Free: 90 GB
+Target: type unknown, state down_out
+- Storage tier 0 (SCM):
+  Total size: 6.0 GB
+  Free: 5.0 GB
+- Storage tier 1 (NVMe):
+  Total size: 100 GB
+  Free: 90 GB
+Target: type unknown, state up_in
+- Storage tier 0 (SCM):
+  Total size: 6.0 GB
+  Free: 5.0 GB
+- Storage tier 1 (NVMe):
+  Total size: 100 GB
+  Free: 90 GB
+`,
```

Comment:
Would be nice to add a test for the case where there is a storage tier 2 (unknown).

Reply:
added

Reply:
> > what are the irrelevant proto updates in this patch? Are you referring to maybe something in the auto-generated code? Or, in the mgmt.proto or pool.proto changes?
> 
> I'm referring to the auto-generated code. You only changed mgmt and pool, but it looks like everything got regenerated. You should be able to revert anything Protobuf generated except the mgmt.pb.* and pool.pb.*.

should be fixed in latest commits